eclipse-theia/theia
# Prevent MonacoEditorService from instantiating ContextKeyService thro…

…ugh Monaco



#### What it does


Fixes: https://github.com/eclipse-theia/theia/issues/11096.

If we wish to override Monaco services, we must do so before the same services are instantiated by Monaco by other means. At the moment, one of our desired overrides, the `MonacoEditorService` is instantiating a `ContextKeyService` using Monaco methods, but we want to override the `ContextKeyService` as well. This PR changes the way the `MonacoEditorService` accesses the `ContextKeyService` to allow the override to take effect.

#### How to test


1. Install [this plugin](https://github.com/vince-fugnitto/enablement-test/releases/tag/0.0.1)
2. Use the command palette to run the `Enablement: Enable` and `Enablement: Disable` commands.
3. Observe that the `Show Enablement Issue` widget appears and disappears from its view container. (But due to https://github.com/eclipse-theia/theia/issues/10778 , the view container will not close)

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
